src/wp-includes/rest-api/endpoints/class-wp-rest-blocks-controller.php

```diff
@@ -58,8 +58,17 @@ public function filter_response_by_context( $data, $context ) {
 		unset( $data['content']['rendered'] );
 
 		// Add the core wp_pattern_sync_status meta as top level property to the response.
-		$data['wp_pattern_sync_status'] = isset( $data['meta']['wp_pattern_sync_status'] ) ? $data['meta']['wp_pattern_sync_status'] : '';
-		unset( $data['meta']['wp_pattern_sync_status'] );
+		$sync_status = '';
+		if ( isset( $data['meta'] ) ) {
+			if ( is_array( $data['meta'] ) && isset( $data['meta']['wp_pattern_sync_status'] ) ) {
+				$sync_status = $data['meta']['wp_pattern_sync_status'];
+				unset( $data['meta']['wp_pattern_sync_status'] );
+			} elseif ( is_object( $data['meta'] ) && isset( $data['meta']->wp_pattern_sync_status ) ) {
+				$sync_status = $data['meta']->wp_pattern_sync_status;
+				unset( $data['meta']->wp_pattern_sync_status );
+			}
+		}
+		$data['wp_pattern_sync_status'] = $sync_status;
```

Comment:
See my overarching feedback: This is a perfect example of why changing the PHP array to an object too early is risky: Here it's an internal situation, but there are filters running over the data, so any external code would need to be updated too. Therefore we need to handle the type change as close as possible before JSON encoding.